### PR TITLE
Added engraving read context (Step 1)

### PR DESCRIPTION
**Pull request by igorkorsukov**
Step by step
Some improvements to the implementation of the engraving domain model

The ultimate goal is to remove the `EngravingObject::score` method, it is very bad, it allows us to wherever do anything.

A step towards this is adding read context.
In fact, the reading context is already there, but not explicitly, the Score and Xml reader was used as the read context.
There will now be an explicit read context
The implementation of the read context refers to Score, but this is optional and may be changed in the future.

Also, the read context shrinks the interface to the minimum, instead of providing a huge Score interface.

### CODE REVIEW

File: src/engraving/accessibility/accessiblescore.cpp [hunk 1]
```diff
@@ -110,7 +110,7 @@ size_t AccessibleScore::accessibleChildCount() const
     for (const EngravingObject* obj : m_score->EngravingObject::children()) {
         if (obj->isEngravingItem()) {
             AccessibleItem* access = Ms::toEngravingItem(obj)->accessible();
-            if (access && access->isAvalaible()) {
+            if (access && access->registred()) {
```
Comment (Marr11317) [inline src/engraving/accessibility/accessiblescore.cpp:113]: registered

File: src/engraving/accessibility/accessiblescore.cpp [hunk 2]
```diff
@@ -125,7 +125,7 @@ const IAccessible* AccessibleScore::accessibleChild(size_t i) const
...
```
Comment (Marr11317) [inline src/engraving/accessibility/accessiblescore.cpp:128]: dito

File: src/engraving/compat/read114.cpp [hunk 3]
```diff
@@ -2828,7 +2829,7 @@ Score::FileError Read114::read114(MasterScore* masterScore, XmlReader& e)
                 if (e.name() == "tempo") {
                     int tick   = e.attribute("tick").toInt();
                     double tmp = e.readElementText().toDouble();
-                    tick       = (tick * MScore::division + masterScore->_fileDivision / 2) / masterScore->_fileDivision;
+                    tick       = ctx.fileDivision(tick);
```
Comment (Marr11317) [inline src/engraving/compat/read114.cpp:2832]: isn't there some calculations lost here?

File: src/engraving/libmscore/ledgerline.cpp [hunk 4]
```diff
@@ -46,6 +46,10 @@ LedgerLine::LedgerLine(Score* s)
     _next       = 0;
 }
 
+LedgerLine::~LedgerLine()
```
Comment (Marr11317) [inline src/engraving/libmscore/ledgerline.cpp:49]: are empty destructors recommended in the style guide?

Comment (igorkorsukov) [reply src/engraving/libmscore/ledgerline.cpp:49]: no, the code style guide doesn't formalize it.
I added an empty destructor to put a breakpoint and see where these objects are being deleted, to see why so many of them are created and deleted during layout
https://github.com/musescore/MuseScore/pull/9100

File: src/engraving/compat/read114.cpp (hunk 3 above)
Comment (igorkorsukov) [reply src/engraving/compat/read114.cpp:2832]: no, it's ok, this method will be called https://github.com/musescore/MuseScore/blob/master/src/engraving/libmscore/score.h#L886

File: src/engraving/accessibility/accessiblescore.cpp (hunk 1 above)
Comment (igorkorsukov) [reply src/engraving/accessibility/accessiblescore.cpp]: Thank you, fixed.

**Review by vpereverzev — APPROVED** [review]

**Pull request merged**